### DISCUSSION TITLE
X Window System error on run

### DISCUSSION THREAD

**Discussion by SekoiaTree** [tauri-apps/tauri#6729] [Q&A] [answered]
When running a built deb image:
```
(WebKitWebProcess:103134): Gdk-ERROR **: 20:32:04.545: The program 'WebKitWebProcess' received an X Window System error.
This probably reflects a bug in the program.
The error was 'BadValue (integer parameter out of range for operation)'.
  (Details: serial 220 error_code 2 request_code 152 (GLX) minor_code 34)
  (Note to programmers: normally, X errors are reported asynchronously;
   that is, you will receive the error a while after causing it.
   To debug your program, run it with the GDK_SYNCHRONIZE environment
   variable to change this behavior. You can then get a meaningful
   backtrace from your debugger if you break on the gdk_x_error() function.)
```
The window just stays gray. I'm fully updated and I've restarted my computer. I am on Pop!_OS 22.04 LTS.

**Comment by FabianLars** [comment, marked as answer]
A few questions from me i asked in thread with a very similar error: https://github.com/tauri-apps/tauri/issues/5085#issuecomment-1230089593
> Errors like this are basically impossible to debug (for us/me) tbh, but a few questions anyway:

> WSL or an actual Linux installation?
    What GPU are you using?
    If AMD: Are you using the open-source drivers (that are part of the kernel) or amdgpu-pro?
    Are all your system dependencies up-to-date?

> Edit: Another thing i saw on the internet, can you try setting this env var: WEBKIT_FORCE_SANDBOX=0 ? Probably doesn't work but worth a try :/

And another thing you can try is `WEBKIT_DISABLE_COMPOSITING_MODE=1` which i saw in https://github.com/tauri-apps/tauri/issues/3407

Would you mind trying these things and answering my questions? Then we can try to figure out the problem, even though i'm not really optimistic tbh.

**Reply by SekoiaTree** [reply]
Actual linux installation. Nvidia RTX 3070. Yes, everything is updated. Yes, I've tried both of those environment variables. Nothing changed, sadly (maybe I used it wrong? I just ran `WEBKIT_DISABLE_COMPOSITING_MODE=1` in the commandline and ran the program again (from the same console), same with FORCE_SANDBOX).

**Reply by FabianLars** [reply]
> I just ran WEBKIT_DISABLE_COMPOSITING_MODE=1 in the commandline and ran the program again (from the same console), same with FORCE_SANDBOX).

If it's 2 separate commands you need to add an `export` like: 
```
export WEBKIT_DISABLE_COMPOSITING_MODE=1
```
alternatively you can combine them into one command like
```
WEBKIT_DISABLE_COMPOSITING_MODE=1 ./your-app
```

**Reply by SekoiaTree** [reply]
Okay, so WEBKIT_DISABLE_COMPOSITING_MODE worked, but SANDBOX didn't

**Reply by rivy** [reply]
@FabianLars , thanks for taking the time to address this concern.

I've had the same problem on Win10 using ssh with a local X11 server to an Ubuntu host in multiple WebKit-based apps.
This fixes the failure but, as noted, causes other graphics degradation.
Is this fundamentally an issue with WebKit? And, if so, where should we post the bug report/issue?

**Reply by FabianLars** [reply]
> Is this fundamentally an issue with WebKit?

I think so, yes. Of course not 100% sure but i can't think of anything that Tauri does that could cause these issues.

> And, if so, where should we post the bug report/issue?

Their bug tracker is over here https://bugs.webkit.org/enter_bug.cgi

**Reply by mberry** [reply]
Was facing this issue trying to debug with vscode and setting `WEBKIT_DISABLE_COMPOSITING_MODE=1` was the fix, haven't noticed any graphics issues since. This is in a container and using X11 Forwarding.

Running tauri normally inside the container via command line worked fine without disabling webkit compositing so I'm not sure what is happening there.

**Comment by Rohit19060** [comment]
Can anyone tell me how to do this in windows? WEBKIT_DISABLE_COMPOSITING_MODE=1

**Reply by FabianLars** [reply]
This is a Linux-only setting.

**Reply by Rohit19060** [reply]
What is the solution for windows for this issue, can we avoid the startup white screen in any way?

**Reply by FabianLars** [reply]
You can spawn the window hidden (visible: false) and then show it at a later point in javascript. Note that if you're using v2 beta you may have to wait until the next release for this to work again.

**Comment by 0rvar** [comment]
Hit this on Ubuntu 22.04 (Wayland + XWayland). Adding
XInitThreads() at the start of main fixed it for us:

```rust
fn main() {
    #[cfg(target_os = "linux")]
    unsafe { x11::xlib::XInitThreads(); }

    // ...
}
```

Never hit this on Ubuntu 24, so not sure what's going on here. But calling XInitThreads should be safe as it's supposed to be idempotent anyway
